### PR TITLE
ci: Limit cache size regardless of NO_DEPENDS

### LINKED ISSUE #18666: Investiage why Bitcoin Core is using terabytes of travis cache space
We are using the travis cache to speed up builds. However, the caches are quite large. They are in the order of 11 GBs for each branch and each pull request, which puts the total usage we put on travis in the order of Terabytes. They don't charge us for cache space, but I believe it still slows down our builds.

Screenshot_2020-04-16 bitcoin bitcoin - Travis CI

Our ccache size is limited to 100 Mb.

https://github.com/bitcoin/bitcoin/blob/e16718a8b3db8bf9c9715f28f4dc6080bf609776/ci/test/00_setup_env.sh#L43

However, some build don't pick up this setting.

https://travis-ci.org/github/bitcoin/bitcoin/jobs/675498487#L3995

#### Useful skills:

Basic understanding of the Bitcoin Core ci system, travis, ccache and our build system.

#### Want to work on this issue?

The purpose of the `good first issue` label is to highlight which issues are suitable for a new contributor without a deep understanding of the codebase.

You do not need to request permission to start working on this. You are encouraged to comment on the issue if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by hebasto**
Close #18666.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
ACK

Depends has ccache disabled anyway and is cached regardless of whether ccache is there or not, see #17248

File: ci/test/06_script_a.sh [hunk 1]
```diff
@@ -8,9 +8,7 @@ export LC_ALL=C.UTF-8
 
 BITCOIN_CONFIG_ALL="--disable-dependency-tracking --prefix=$DEPENDS_DIR/$HOST --bindir=$BASE_OUTDIR/bin --libdir=$BASE_OUTDIR/lib"
 DOCKER_EXEC "command -v ccache > /dev/null && ccache --zero-stats"
-if [ -z "$NO_DEPENDS" ]; then
-  DOCKER_EXEC ccache --max-size=$CCACHE_SIZE
-fi
+DOCKER_EXEC "command -v ccache > /dev/null && ccache --max-size=$CCACHE_SIZE"
```
Comment (MarcoFalke) [inline ci/test/06_script_a.sh:11]:
```suggestion
DOCKER_EXEC "ccache --max-size=$CCACHE_SIZE"
```

Ccache is a requirement, why would this not be there?

Comment (hebasto) [reply ci/test/06_script_a.sh:11]: > Ccache is a requirement for ci builds...

Yes, it is now. But it seems not guaranteed for custom defined `DOCKER_PACKAGES` macro:
https://github.com/bitcoin/bitcoin/blob/661e8df1b63b213d2d5b0d7cba0733869b508af9/ci/test/00_setup_env.sh#L56

Comment (hebasto) [reply ci/test/06_script_a.sh:11]: Maybe it is worth to disable ability to provide custom `DOCKER_PACKAGES` value as we have the `PACKAGES` macro on that purpose, no?

**Comment by hebasto** [comment]
Updated 4f29a488f19a5eb7ba21f4a9cce5a5fe9a4d8d3d  -> 087e7a3b1dd97094792a286ee34c69b9c2f37fc2 ([pr18667.01](https://github.com/hebasto/bitcoin/commits/pr18667.01) -> [pr18667.02](https://github.com/hebasto/bitcoin/commits/pr18667.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18667.01..pr18667.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18667#discussion_r409593544):
> Ccache is a requirement for ci builds, why would this not be there?

File: ci/test/06_script_a.sh (hunk 1 above)
Comment (MarcoFalke) [reply ci/test/06_script_a.sh]: I still like the ability to provide a custom DOCKER_PACKAGES. `ccache` is required for the ci in the same way that a compiler or python is required. Without it, it doesn't run.

**Comment by hebasto** [comment]
Updated 087e7a3b1dd97094792a286ee34c69b9c2f37fc2  -> 0c6318788beaf1a31aeba5a21f3f8bb5c07cea6c ([pr18667.02](https://github.com/hebasto/bitcoin/commits/pr18667.02) -> [pr18667.03](https://github.com/hebasto/bitcoin/commits/pr18667.03), [diff](https://github.com/hebasto/bitcoin/compare/pr18667.02..pr18667.03)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18667#discussion_r409619427):
> I still like the ability to provide a custom DOCKER_PACKAGES. `ccache` is required for the ci in the same way that a compiler or python is required. Without it, it doesn't run.

File: ci/test/06_script_a.sh (hunk 1 above)
Comment (hebasto) [reply ci/test/06_script_a.sh]: [Updated](https://github.com/bitcoin/bitcoin/pull/18667#issuecomment-614706768).

**Pull request merged**